### PR TITLE
add sudo call to set duration for schedule call

### PR DESCRIPTION
**Pull request by open-junius**
## Related Issue(s)

- Closes #[issue number]

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

## Breaking Change

If this PR introduces a breaking change, please provide a detailed description of the impact and the migration path for existing applications.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please include any relevant screenshots or GIFs that demonstrate the changes made.

## Additional Notes

Please provide any additional information or context that may be helpful for reviewers.

### CODE REVIEW

**Review by sam0x17 — DISMISSED** [review]

**Review by sam0x17 — DISMISSED** [review]

**Review by sam0x17 — DISMISSED** [review]

File: pallets/subtensor/src/coinbase/root.rs [hunk 1]
```diff
@@ -992,9 +992,8 @@ impl<T: Config> Pallet<T> {
     /// * 'SubNetworkDoesNotExist': If the specified network does not exist.
     /// * 'NotSubnetOwner': If the caller does not own the specified subnet.
     ///
-    pub fn user_remove_network(origin: T::RuntimeOrigin, netuid: u16) -> dispatch::DispatchResult {
+    pub fn user_remove_network(coldkey: T::AccountId, netuid: u16) -> dispatch::DispatchResult {
         // --- 1. Ensure the function caller is a signed user.
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/coinbase/root.rs]: fix comments

File: pallets/subtensor/src/macros/dispatches.rs [hunk 2]
```diff
@@ -1100,7 +1105,10 @@ mod dispatches {
             let duration: BlockNumberFor<T> = DissolveNetworkScheduleDuration::<T>::get();
             let when: BlockNumberFor<T> = current_block.saturating_add(duration);
 
-            let call = Call::<T>::dissolve_network { netuid };
+            let call = Call::<T>::dissolve_network {
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/macros/dispatches.rs:1108]: Do we need to check that the user owns the network they are trying to dissolve here ? From the looks of it , anyone can dissolve a network , so I think we should just check if the user is the owner of the network

File: pallets/subtensor/src/macros/dispatches.rs [hunk 3]
```diff
@@ -681,7 +681,7 @@ mod dispatches {
             new_coldkey: T::AccountId,
         ) -> DispatchResultWithPostInfo {
             // Ensure it's called with root privileges (scheduler has root privileges)
-            ensure_root(origin.clone())?;
+            ensure_root(origin)?;
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/macros/dispatches.rs:684]: nevermind , the check is the method

**Review by distributedstatemachine — DISMISSED** [review]

File: pallets/subtensor/src/macros/dispatches.rs (hunk 2 above)
Comment (open-junius) [reply pallets/subtensor/src/macros/dispatches.rs:1108]: The owner could be updated after the call scheduled. You can check my test case. We could let the dissolve_network extrinsic to check when it is executed.

File: pallets/subtensor/src/coinbase/root.rs (hunk 1 above)
Comment (open-junius) [reply pallets/subtensor/src/coinbase/root.rs]: will fix it.

**Review by unconst — APPROVED** [review]

**Pull request merged**